.github/workflows/docs-build.yaml

```diff
@@ -36,14 +37,22 @@ jobs:
       - name: Detect change
         id: change
         run: |
-          first_commit=$(git rev-list HEAD | tail -n 1)
-          diff=$(git diff --name-only HEAD..${first_commit})
-          if [[ "${diff}" =~ docker/Dockerfile ]] && [[ "${{ github.event_name }}" == 'pull_request' ]]; then
+          echo "${{ secrets.GITHUB_TOKEN }}" | docker login -u "${{ github.actor }}" --password-stdin "${{env.REGISTRY}}"
+          if ! docker manifest inspect "${REGISTRY}/${IMAGE_NAME}:${TAG}" 2>&1 > /dev/null ; then export NEEDS_IMAGE=true ; fi
+          def_branch=$(gh api "repos/${GITHUB_REPOSITORY}" -q '.default_branch')
+          git fetch origin "${def_branch}"
+          files=$(git diff --name-only "${GITHUB_SHA}" "origin/${def_branch}" | tr '\n' ' ')
+          if echo "${files}" | grep -q "deps/\|Dockerfile\|repo.toml"; then export NEEDS_IMAGE=true ; fi
+          if [[ "${NEEDS_IMAGE}" ]]; then
             echo "changed=true" >> "$GITHUB_OUTPUT"
-            echo "image=${REGISTRY}/${IMAGE_NAME}:${{ github.event.pull_request.number }}" >> "$GITHUB_OUTPUT"
+            if [[ "${{ github.event_name }}" == 'pull_request' ]]; then
+              echo "image=${REGISTRY}/${IMAGE_NAME,,}:${{ github.event.pull_request.number }}" >> "$GITHUB_OUTPUT"
+            else
+              echo "image=${REGISTRY}/${IMAGE_NAME,,}:${TAG}" >> "$GITHUB_OUTPUT"
```

Comment:
the image is being tagged `4` , is better to use 
```bash
echo "COMMIT_SHORT_SHA=${GITHUB_SHA:0:8}" >> $GITHUB_ENV
```
so each image related to the commit sha

Reply:
I noticed that practice in the DRA repo.  My thinking is that--as a docs repo rather than software repo--I don't need to distinguish the build container by SHA and identifying them by PR is sufficient.  (There's no traceability back to passed and failed tests and what not.)

Reply:
Still, let's define a better tag than `4` . 
maybe the date or simply `main`

Reply:
I yield.  I was planning to switch to the branch name, but then I'd have to sanitize it...not worth the effort.  Why eight hexadecimal digits are better than a small integer, I'll never know.